test

```diff
@@ -77,6 +90,60 @@ function integration_pass {
+	go list -f 'go test -covermode atomic -coverprofile $COVEROUT/unit_{{.Name}}.coverprofile ${RACE} -cpu 1,2,4 -run=Test {{.ImportPath}}' $TEST | xargs -t -I {} bash -c {}
+	gocovmerge "$COVEROUT"/*.coverprofile > "$COVEROUT"/unit.out
+	rm "$COVEROUT"/*.coverprofile
+	echo "unit test code coverage $(go tool cover -func="$COVEROUT/unit.out" | tail -n1 | awk '{print $(NF)}')"
+}
+
+function integration_cov {
+	echo "Running code coverage for integration tests..."
+	# integration tests must use atomic covermode to ensure additional coverage logic inserted to the source code by go cover does not induce data race due to the multi-threaded nature of integration tests
+	go test -covermode atomic -coverprofile "$COVEROUT/integration.coverprofile" -coverpkg $PKGS_DELIM -timeout 15m -v -cpu 1,2,4 $@ ${REPO_PATH}/integration
+	go test -covermode atomic -coverprofile "$COVEROUT/client_integration.coverprofile" -coverpkg $PKGS_DELIM -timeout 1m -v ${RACE} -cpu 1 $@ ${REPO_PATH}/client/integration
+	go test -covermode atomic -coverprofile "$COVEROUT/clientv3_integration.coverprofile" -coverpkg $PKGS_DELIM -timeout 10m -v ${RACE} -cpu 1,2,4 $@ ${REPO_PATH}/clientv3/integration
+	go test -covermode atomic -coverprofile "$COVEROUT/contrib_raftexample.coverprofile" -coverpkg $PKGS_DELIM -timeout 1m -v -cpu 1,2,4 $@ ${REPO_PATH}/contrib/raftexample
+	# go list (...) | xargs (...) issues following commands where each command runs a unit test against the entire etcd pkgs and generates its coverage report, e.g $COVEROUT/main.coverprofile for github.com/coreos/etcd
+	# 	bash -c go test -covermode atomic -coverprofile $COVEROUT/main.coverprofile ${RACE} -coverpkg $PKGS_DELIM -cpu 1,2,4 -run=Test github.com/coreos/etcd
+	# 	ok  	github.com/coreos/etcd	0.019s	coverage: 0.0% of statements
+	#   ...
+	go list -f 'go test -covermode atomic -coverprofile $COVEROUT/integration_{{.Name}}.coverprofile ${RACE} -coverpkg $PKGS_DELIM -cpu 1,2,4 -run=Example {{.ImportPath}}' $TEST | xargs -t -I {} bash -c {}
+	gocovmerge "$COVEROUT"/*.coverprofile > "$COVEROUT"/integration.out
+	rm "$COVEROUT"/*.coverprofile
+	echo "integration test code coverage $(go tool cover -func="$COVEROUT/integration.out" | tail -n1 | awk '{print $(NF)}')"
+}
+
+function cov_pass {
+	echo "Running code coverage..."
+	# install gocovmerge before running code coverage from github.com/wadey/gocovmerge
+	# gocovmerge merges coverage files 
+	if ! which gocovmerge >/dev/null; then
+    	echo "gocovmerge not installed"
+    	exit 255
+	fi
+	unit_cov &
+	unitpid="$!"
+	integration_cov &
+	intpid="$!"
+	wait $unitpid
+	wait $intpid
+	coverage_all
+}
+
+function coverage_all {
+	gocovmerge "$COVEROUT"/*.out > "$COVEROUT"/all.out
+	echo "total coverage: $(go tool cover -func=$COVEROUT/all.out | tail -n1 | awk '{print $(NF)}')"
+}
+
+# TODO enable e2e coverage test
```

Comment:
@fanminshi Then don't we need something like https://github.com/coreos/etcd/blob/master/.travis.yml#L41-L44